src/ContainerBuilder.php

```diff
@@ -102,4 +102,38 @@ public function newConfiguredInstance(
 
         return $di;
     }
+
+    /**
+     *
+     * Get config object from connfig class or return the object
+     *
+     * @param mixed $config name of class to instantiate
+     *
+     * @return Object
+     * @throws InvalidArgumentException if invalid config
+     *
+     * @access protected
+     */
+    protected function getConfig($config)
+    {
+        if (is_string($config)) {
+            $config = new $config;
+        }
+
+        if (! is_object($config)) {
+            throw new \InvalidArgumentException(
+                'Container configs must be class names or objects'
+            );
+        }
+
+        if (! method_exists($config, 'define')
```

Comment:
I don't really understand your reasoning of checking with instance of `ContainerConfigInterface` . May be you are saying at some point of time the container interoperability will have the interface it will break ?